bot.py

```diff
@@ -321,12 +320,9 @@ def msg_ping(match: Match[str]) -> Response:
 @handle_message(r'.*\b(nano|linux|windows|emacs)\b', flags=re.IGNORECASE)
 def msg_gnu_please(match: Match[str]) -> Response:
     msg, word = match[3], match[4]
-    cases = {
-        f'{gnu}{sep}{word}'
-        for gnu, sep in itertools.product(('gnu', 'GNU'), '/+')
-    }
-    if msg in cases:
-        return MessageResponse(match, f'YES! GNU/{esc(word)}')
+    query = re.match(fr'(?i)gnu(\/|\+){word}', msg)
```

Comment:
- can move `(?i)` to the third argument as `re.I`
- can use `[/+]` instead of the `(...)`